src/crypto.js

```diff
@@ -0,0 +1,266 @@
+import { generateError, ERRORS, checkObject } from './errors'
+
+const genRandomBuffer = (len = 16) => {
```

Comment:
This does not return a buffer right ?

Reply:
Right, it returns a int8Array, we can use Buffer.from(returned Array from genRandomBuffer) if necessary.